web/app.js

```diff
@@ -1835,6 +1854,8 @@ function webViewerUpdateViewarea(evt) {
       'scrollLeft': location.left,
       'scrollTop': location.top,
       'rotation': location.rotation,
+      'scrollMode': location.scrollMode,
+      'spreadMode': location.spreadMode,
```

Comment:
While working on this and comparing with the sidebar behavior, I noticed that the `sidebarViewOnLoad` preference, if set, overrides what a user has previously chosen for the document. Is this correct, and should the scroll/spread mode preferences also do this?

Reply:
During my tests, the user-set preference is only ignored if the user has changed it to the default value. This seems odd, and worthy of a separate bug report (I guess that it is caused by a `|| ...` clause somewhere).

The value of scroll/spread mode may significantly affect the layout of the page, so the (saved) document-specific preference should take precedence over the general user preferences. I think that currently the document-specific preference is already modified if the value differs from the source defaults (instead of the default as set by the general user preferences). This is fine.
(An alternative is to only save the document-specific preference if the value differs from the generic user preference, and to remove the document-specific preference if the value matches the default, but this leads to more complicated code without large benefits.)